src/PlaylistController.js

```diff
@@ -0,0 +1,103 @@
+import assert from 'assert';
+import GenericController from './GenericController';
+import Playlist from './Structs/Playlist';
+
+export default class PlaylistController extends GenericController {
+  constructor(...args) {
+    super(...args);
+    if (window.APPCONTEXT && window.APPCONTEXT.Go && window.APPCONTEXT.Go.h && window.APPCONTEXT.Go.h.length) {
+      this._playlists = Object.assign({}, window.APPCONTEXT.Go.h[0].Hi);
+      this._watchPlaylistObject();
+    } else {
+      this._playlists = {};
+    }
+
+    this.addMethod('getAll', this.getAll.bind(this));
+    this.addMethod('play', this.play.bind(this));
+    this.addMethod('playWithTrack', this.playWithTrack.bind(this));
+  }
+
+  _navigate(playlist) {
+    return new Promise((resolve, reject) => {
+      window.location.hash = `/pl/${escape(playlist.id)}`;
+      let waitForPage;
+      const waitTimeout = setTimeout(() => clearInterval(waitForPage) && reject('Playlist took too long to load, it might not exist'), 10000);
```

Comment:
Ohhh, I see now. That is confusing x_x Maybe they should both be a `let` and we have a function to stop timeouts:

```
let waitForPage, waitTimeout;
clearTimeouts() => { clearInterval(waitForPage); clearTimeout(waitTimeout) };

waitForPage = setInterval() = {
  // ...
  clearTimeouts();
}
```